File: Haskell/FizzBuzz/FizzBuzz.hs
module FizzBuzz where

fizzBuzzRep :: Int -> [Char]
listToFizzBuzz :: [Int] -> [[Char]]
printFizzBuzz :: [Int] -> IO ()

fizzBuzzRep x
    | x `mod` 5 == 0 && x `mod` 3 == 0 = "FizzBuzz"
    | x `mod` 5 == 0                   = "Buzz"
    | x `mod` 3 == 0                   = "Fizz"
    | otherwise                        = show x

listToFizzBuzz xs = [fizzBuzzRep x | x <- xs]

printFizzBuzz (x:xs) = do putStrLn (fizzBuzzRep x)
                          printFizzBuzz xs

printFizzBuzz _ = putStrLn ""

main = printFizzBuzz [1..100]

File: JavaScript/FizzBuzz/fizzBuzz.js
fizzBuzzRep = function (number) {
  var result = "";

  if (number % 3 == 0) {
    result += "Fizz";
  }
  if (number % 5 == 0) {
    result += "Buzz";
  }

  return result || number.toString();
};

getFizzBuzzRange = function (low, high) {
  var result = [],
      i;

  for (i = 0; i <= high-low; i += 1) {
    result.push(fizzBuzzRep(i+low));
  }

  return result;
};
